### PR TITLE
Fix release build and fbcode+clang+shared

### LINKED ISSUE #7061: recompile error
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://groups.google.com/forum/#!forum/rocksdb or https://www.facebook.com/groups/rocksdb.dev

### Expected behavior
compile correctly
### Actual behavior
compile error:
```
  CC       third-party/folly/folly/synchronization/WaitOptions.o
  AR       librocksdb.a
ar: creating librocksdb.a
  CCLD     db_stress
db_stress_tool/db_stress_test_base.o: In function `rocksdb::StressTest::TestCheckpoint(rocksdb::ThreadState*, std::vector<int, std::allocator<int> > const&, std::vector<long, std::allocator<long> > const&)':
/data/rocksdb/db_stress_tool/db_stress_test_base.cc:1343: undefined reference to `rocksdb::test::DestroyDir(rocksdb::Env*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
Makefile:1215: recipe for target 'db_stress' failed
make[1]: *** [db_stress] Error 1
make[1]: Leaving directory '/data/rocksdb'
Makefile:711: recipe for target 'release' failed
make: *** [release] Error 2
```
### Steps to reproduce the behavior
1.  first I run `cd rocksdb && CFLAGS=-fPIC CXXFLAGS=-fPIC make release` compiled correctly.
2. some other reasons I run `rm -rf rocksdb && git clone https://github.com/facebook/rocksdb && cd rocksdb && CFLAGS=-fPIC CXXFLAGS=-fPIC make release` error like above. 

system: Ubuntu 16.04.4 LTS
how to solve it?  thanks.

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: Follow-up to #6660. Release build had linker error. fbcode+clang+shared build was erroring on unused parameter '-nostdinc'.

Fixes #7061 

Test Plan: make release, USE_CLANG=1 LIB_MODE=shared make check, etc.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1211,7 +1211,7 @@ filter_bench: $(OBJ_DIR)/util/filter_bench.o $(LIBRARY)
 	$(AM_LINK)
 
 ifeq ($(DEBUG_LEVEL),0)
-db_stress: $(OBJ_DIR)/db_stress_tool/db_stress.o $(STRESS_OBJECTS) $(LIBRARY)
+db_stress: $(OBJ_DIR)/db_stress_tool/db_stress.o $(STRESS_OBJECTS) $(TESTUTIL) $(LIBRARY)
```
Comment (mrambacher) [inline Makefile:1214]: Can you explain why this change?

Comment (pdillinger) [reply Makefile:1214]: Fix release build. I think it's more implicitly understandable than much of the Makefile. If you remove it and "cover" the line in your build, you're guaranteed a linker error. And

db_stress_tool/db_stress_common.h:72:#include "test_util/testutil.h"

A TODO item from this is to cover the release build in CI

Comment (pdillinger) [reply Makefile:1214]: > A TODO item from this is to cover the release build in CI

Actually, we have it. https://app.circleci.com/pipelines/github/facebook/rocksdb/2037/workflows/7da6367b-5f7b-4ce9-9c15-8aa65c746266/jobs/4511  I have no idea why it passed in CircleCI because it fails for me every way I try "make release" on that revision
